fastly/block_fastly_service_v1_logging_datadog.go

```diff
@@ -0,0 +1,210 @@
+			},
+
+			"format": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Apache-style string or VCL variables to use for log formatting.",
+			},
+
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
+				ValidateFunc: validateLoggingPlacement(),
+			},
+
+			"response_condition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the condition to apply",
+			},
+		},
+	},
+}
+
+func processDatadog(d *schema.ResourceData, conn *fst.Client, latestVersion int) error {
+	serviceID := d.Id()
+	od, nd := d.GetChange("logging_datadog")
+
+	if od == nil {
+		od = new(schema.Set)
+	}
+	if nd == nil {
+		nd = new(schema.Set)
+	}
+
+	ods := od.(*schema.Set)
+	nds := nd.(*schema.Set)
+
+	removeDatadogLogging := ods.Difference(nds).List()
+	addDatadogLogging := nds.Difference(ods).List()
+
+	// DELETE old Datadog logging endpoints
+	for _, oRaw := range removeDatadogLogging {
+		of := oRaw.(map[string]interface{})
+		opts := buildDeleteDatadog(of, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly Datadog logging endpoint removal opts: %#v", opts)
+
+		if err := deleteDatadog(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	// POST new/updated Datadog logging endpoints
+	for _, nRaw := range addDatadogLogging {
+		df := nRaw.(map[string]interface{})
+		opts := buildCreateDatadog(df, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly Datadog logging addition opts: %#v", opts)
+
+		if err := createDatadog(conn, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func readDatadog(conn *fst.Client, d *schema.ResourceData, s *fst.ServiceDetail) error {
+	// refresh Datadog
+	log.Printf("[DEBUG] Refreshing Datadog logging endpoints for (%s)", d.Id())
+	datadogList, err := conn.ListDatadog(&fst.ListDatadogInput{
+		Service: d.Id(),
+		Version: s.ActiveVersion.Number,
+	})
+
+	if err != nil {
+		return fmt.Errorf("[ERR] Error looking up Datadog logging endpoints for (%s), version (%v): %s", d.Id(), s.ActiveVersion.Number, err)
+	}
+
+	dll := flattenDatadog(datadogList)
+
+	if err := d.Set("logging_datadog", dll); err != nil {
+		log.Printf("[WARN] Error setting Datadog logging endpoints for (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func createDatadog(conn *fst.Client, i *fst.CreateDatadogInput) error {
+	_, err := conn.CreateDatadog(i)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteDatadog(conn *fst.Client, i *fst.DeleteDatadogInput) error {
+	err := conn.DeleteDatadog(i)
+
+	if errRes, ok := err.(*fst.HTTPError); ok {
```

Comment:
Same as other PRs, but still noting, wouldn't this result in 404s returning nil? Is this expected?